src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -64,14 +64,30 @@ export function DiscoverSidebar({
   useNewFieldsApi = false,
   useFlyout = false,
   unmappedFieldsConfig,
+  onEditRuntimeField,
 }: DiscoverSidebarProps) {
   const [fields, setFields] = useState<IndexPatternField[] | null>(null);
-
+  const { indexPatternFieldEditor } = services;
+  const indexPatternFieldEditPermission = indexPatternFieldEditor?.userPermissions.editIndexPattern();
+  const canEditIndexPatternField = !!indexPatternFieldEditPermission && useNewFieldsApi;
   useEffect(() => {
     const newFields = getIndexPatternFieldList(selectedIndexPattern, fieldCounts);
     setFields(newFields);
   }, [selectedIndexPattern, fieldCounts, hits]);
 
+  const closeFieldEditor = useRef<() => void | undefined>();
+  useEffect(() => {
+    const cleanup = () => {
+      if (closeFieldEditor.current) {
+        closeFieldEditor.current();
+      }
+    };
+    return () => {
+      // Make sure to close the editor when unmounting
+      cleanup();
+    };
+  }, []);
```

Comment:
I think `closeFieldEditor` is never assigned?

Reply:
Good catch!